runtime/reconcilers/util.go

```diff
@@ -154,3 +155,48 @@ func safeSQLName(name string) string {
 	}
 	return fmt.Sprintf("\"%s\"", strings.ReplaceAll(name, "\"", "\"\""))
 }
+
+func resolveTemplateProps(ctx context.Context, props map[string]any, self compilerv1.TemplateResource, c *runtime.Controller) (map[string]any, error) {
+	inst, err := c.Runtime.Instance(ctx, c.InstanceID)
+	if err != nil {
+		return nil, err
+	}
+	vars := inst.ResolveVariables()
+
+	for key, value := range props {
+		strValue, ok := value.(string)
+		if !ok {
+			continue
+		}
+
+		strValue, err = compilerv1.ResolveTemplate(strValue, compilerv1.TemplateData{
```

Comment:
Two issues:
1. It doesn't traverse nested properties (nested lists/maps)
2. It creates a new `compilerv1.TemplateData` on each iteration – can just be created once

Reply:
Looked at the internal implementation of `propsPB.AsMap()` and maps and lists seems to be of type `map[string]any` and `[]any`. Adding specific handling for these types to avoid `reflection`.